src/classes/Api.php

```diff
@@ -160,4 +167,29 @@ public function reset() {
             $this->adapter = NULL;
         }
 
+        /**
+         * @inheritdoc
+         */
+        public function getFormFields($recipientlistId)
+        {
+
+            $cacheFilePath = get_temp_dir() . 'rapidmail_fields.php';
+            $cacheData = null;
+
+            if (!is_file($cacheFilePath) || (time() - filemtime($cacheFilePath)) > self::API_FIELDS_CACHE_MAXAGE_SECONDS) {
+
+                $cacheData = $this->adapter()->getFormFields($recipientlistId);
+
+                if (false === file_put_contents($cacheFilePath, '<?php return '. var_export($cacheData, true) . ';')) {
+                    throw new \RuntimeException('File "' . $cacheFilePath . '" could not be written');
+                }
+
+            } else {
+                $cacheData = require $cacheFilePath;
```

Comment:
Might be a security issue if someone else puts evil code into TEMP_DIR/rapidmail_fields.php`

Reply:
Changed to JSON.